R/tt_compatibility.R

```diff
@@ -374,6 +374,25 @@ rbindl_rtables <- function(x, gap = 0, check_headers = TRUE) {
 
     lapply(x, function(xi) chk_compat_cinfos(x[[1]], xi)) ##col_info(xi)))
 
+    rbind_annot <- list(title = "", subtitles = character(), footer = character(), pf = character())
+    # Titles/footer info are (independently) retained from first object if all other objects have none
+    all_titles <- lapply(x, function(x) if (is.null(attr(x, "main_title"))) "" else main_title(x))
+    if (all_titles[[1]] != "" & all(all_titles[[-1]] == "")) rbind_annot[["title"]] <- all_titles[[1]]
+
+    all_sts <- lapply(x, function(x) if (is.null(attr(x, "subtitles"))) character() else subtitles(x))
+    if (length(all_sts[[1]]) > 0 & all(sapply(all_sts[[-1]], function(x) length(x) == 0))) {
+      rbind_annot[["subtitles"]] <- all_sts[[1]]
+    }
+
+    all_footers <- lapply(x, function(x) if (is.null(attr(x, "main_footer"))) character() else main_footer(x))
+    if (length(all_footers[[1]]) > 0 & all(sapply(all_footers[[-1]], function(x) length(x) == 0))) {
```

Comment:
Overall, I think retaining annotations should be an all or nothing thing. If any of the annotations disagree between two fo the tables, they no annotations should be perserved. If they ALL agree, or if all but the first table have no annotations at all, then the first table's annotations should be retained.